### PR TITLE
Wrap quickpick and input box results into options

### PR DESCRIPTION
**Pull request by kpodsiad**
Fixes https://github.com/scalameta/metals/issues/3338
Should we add some tests for it?

### CODE REVIEW

**Review by dos65 — APPROVED** [review]
LGTM!

What about tests, I don't think that it is worth covering by a test every case where we ask input/items that just to check that it handles empty response correctly. Precise types do this job much better.

**Review by tgodzik — CHANGES_REQUESTED** [review]
Just a small issue, otherwise looks good!

File: metals/src/main/scala/scala/meta/internal/metals/findfiles/FindTextInDependencyJars.scala [hunk 1]
```diff
@@ -168,16 +168,9 @@ class FindTextInDependencyJars(
         languageClient
           .metalsInputBox(input)
           .asScala
-          .map(checkResult)
+          .mapOptionInside(_.value)
```
[the remark this answers is not in the document]
Comment (kpodsiad) [reply metals/src/main/scala/scala/meta/internal/metals/findfiles/FindTextInDependencyJars.scala]: Yeah, good catch!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by ckipp01** [comment]
Sorry I missed this, but 
> Should we add some tests for it?

Absolutely. This broke both the new file provider and the new project provider. Gson can't handle the scala types so when you attempt to parse the result you end up with 

```
SEVERE: Failed to invoke public scala.Option() with no args
java.lang.RuntimeException: Failed to invoke public scala.Option() with no args
        at com.google.gson.internal.ConstructorConstructor$3.construct(ConstructorConstructor.java:113)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:212)
        at com.google.gson.Gson.fromJson(Gson.java:963)
        at com.google.gson.Gson.fromJson(Gson.java:1034)
        at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.fromJson(MessageTypeAdapter.java:339)
        at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.parseResult(MessageTypeAdapter.java:221)
        at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.read(MessageTypeAdapter.java:135)
        at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.read(MessageTypeAdapter.java:55)
        at com.google.gson.Gson.fromJson(Gson.java:963)
        at org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler.parseMessage(MessageJsonHandler.java:119)
        at org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler.parseMessage(MessageJsonHandler.java:114)
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:193)
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
        at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.InstantiationException
        at java.base/jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at com.google.gson.internal.ConstructorConstructor$3.construct(ConstructorConstructor.java:110)
        ... 18 more
```
